api/coverage.py

```diff
@@ -240,15 +240,21 @@ def _process_batch(self, client_method, success_codes, batch):
         for message in self.process_feed_response(response, id_mapping):
             try:
                 identifier, _new = Identifier.parse_urn(self._db, message.urn)
-                mapped_batch.remove(identifier)
+                if identifier in mapped_batch:
+                    mapped_batch.remove(identifier)
             except ValueError as e:
                 # For some reason this URN can't be parsed. This
                 # shouldn't happen.
                 continue
 
             if message.status_code in success_codes:
-                result = id_mapping[identifier]
-                results.append(result)
+                if identifier in id_mapping:
+                    result = id_mapping[identifier]
+                    results.append(result)
+                else:
+                    # The server sent information about an identifier
+                    # we didn't ask for. Do nothing.
+                    pass
```

Comment:
I know why these changes are necessary but not why they only recently caused tests to fail -- I don't think any of the changes made in core recently would have triggered these edge cases.

Reply:
This is so strange!